### PR TITLE
[Fluent][Ready For Review] Vm extensions, child and external child resource supports

### PR DESCRIPTION
**Pull request by anuchandy**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@anuchandy__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (anuchan). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by hovsepm** [comment]
@anuchandy please pull from upstream and resolve merge conflicts.

File: src/ResourceManagement/Compute/Microsoft.Azure.Management.V2.Compute/Domain/IVirtualMachineExtension.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
+/**
+* Copyright (c) Microsoft Corporation. All rights reserved.
+* Licensed under the MIT License. See License.txt in the project root for
```
Comment (hovsepm) [inline src/ResourceManagement/Compute/Microsoft.Azure.Management.V2.Compute/Domain/IVirtualMachineExtension.cs]: please change this to C# style license header.

File: src/ResourceManagement/Resource/Microsoft.Azure.Management.V2.Resource/Domain/Core/IExternalChildResource.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+﻿using Microsoft.Azure.Management.V2.Resource.Core.ResourceActions;
+/**
```
Comment (hovsepm) [inline src/ResourceManagement/Resource/Microsoft.Azure.Management.V2.Resource/Domain/Core/IExternalChildResource.cs]: Please put the license header before anything else in the file.

File: src/ResourceManagement/Resource/Microsoft.Azure.Management.V2.Resource/Core/Wrapper.cs [hunk 3]
```diff
@@ -1,4 +1,6 @@
-﻿namespace Microsoft.Azure.Management.V2.Resource.Core
+﻿using System;
```
Comment (hovsepm) [inline src/ResourceManagement/Resource/Microsoft.Azure.Management.V2.Resource/Core/Wrapper.cs]: Please add the license header.

Comment (anuchandy) [inline src/ResourceManagement/Resource/Microsoft.Azure.Management.V2.Resource/Core/Wrapper.cs]: Thanks @hovsepm, added/updated license as you mentioned

**Comment by anuchandy** [comment]
Related PR specific to child and external chld resource in java https://github.com/Azure/azure-sdk-for-java/pull/1095

**Pull request merged**